modules/sugar/src/main/ts/ephox/sugar/api/node/ShadowDom.ts

```diff
@@ -70,3 +70,31 @@ export const getShadowRoot = (e: Element<DomNode>): Option<Element<ShadowRoot>>
  */
 export const getShadowHost = (e: Element<ShadowRoot>): Element<DomElement> =>
   Element.fromDom(e.dom().host);
+
+/**
+ * When Events bubble up through a ShadowRoot, the browser changes the target to be the shadow host.
+ * This function gets the "original" event target if possible.
+ * This only works if the shadow tree is open - if the shadow tree is closed, event.target is returned.
+ * See: https://developers.google.com/web/fundamentals/web-components/shadowdom#events
+ */
+export const getOriginalEventTarget = (event: Event): Option<EventTarget> => {
+  if (isSupported()) {
+    // When target element is inside Shadow DOM we need to take first element from composedPath
+    // otherwise we'll get Shadow Root parent, not actual target element.
+    // TODO: update @ephox/dom-globals to include Event.composedPath
```

Comment:
This TODO is incorrect. We just need to update the version of dom-globals being used, as dom-globals is not meant to differ from the TS version. The JIRA to upgrade dom-globals is TINY-3312.

Reply:
The main problem is to upgrade dom-globals, is that we need to upgrade every repo 😞

Reply:
I don't think it's incorrect. If you update dom-globals and that means it will include Event.composedPath... well, that's what the comment says.

Reply:
It is though, as we shouldn't update dom-globals to include this, instead we need to update dom-globals itself. The two are very different operations.

Reply:
I've updated the comment to clarify. Ta.